nixos/modules/services/web-apps/filesender.nix

```
@@ -0,0 +1,261 @@
+{ config
+, lib
+, pkgs
+, ...
+}:
+let
+  format = pkgs.formats.php { finalVariable = "$config"; };
+
+  cfg = config.services.filesender;
+  simpleSamlCfg = config.services.simplesamlphp.filesender;
+  fpm = config.services.phpfpm.pools.filesender;
+
+  phpPackage = pkgs.php.withExtensions (
+    { enabled
+    , all
+    }: with all;
+    [
+      xml
+      mbstring
+    ]
+    ++ enabled
+  );
+
+  filesenderConfigPhpFile = format.generate "config.php" cfg.settings;
+
+  filesenderConfigDirectory = pkgs.stdenv.mkDerivation {
+    name = "filesender-config";
+    phases = [ "buildtree" ];
+    buildtree = ''
+      mkdir $out
+      cp ${filesenderConfigPhpFile} $out/config.php
+    '';
+  };
+in
+{
+  meta = {
+    maintainers = with lib.maintainers; [ nhnn ];
+  };
+
+  options.services.filesender = with lib; {
+    enable = mkEnableOption "Filesender";
+    package = mkPackageOption pkgs "filesender" { };
+    user = lib.mkOption {
+      description = ''
+        User under which filesender runs.
+      '';
+      type = lib.types.str;
+      default = "filesender";
+    };
+    database = {
+      createLocally = mkOption {
+        type = types.bool;
+        default = true;
+        description = ''
+          Create the PostgreSQL database and database user locally.
+        '';
+      };
+      hostname = mkOption {
+        type = types.str;
+        default = "localhost";
+        description = "Database hostname.";
+      };
+      port = mkOption {
+        type = types.port;
+        default = 5432;
+        description = "Database port.";
+      };
+      name = mkOption {
+        type = types.str;
+        default = "filesender";
+        description = "Database name.";
+      };
+      user = mkOption {
+        type = types.str;
+        default = "filesender";
+        description = "Database user.";
+      };
+      passwordFile = mkOption {
+        type = types.nullOr types.path;
+        default = null;
+        example = "/run/keys/filesender-dbpassword";
+        description = ''
+          A file containing the password corresponding to
+          [](#opt-services.filesender.database.user).
+        '';
+      };
+    };
+    settings = mkOption {
+      type = lib.types.submodule {
+        freeformType = format.type;
+        options = {
+          site_url = mkOption {
+            type = types.str;
+            description = "Site URL. Used in emails, to build URLs for logging in, logging out, build URL for upload endpoint for web workers, to include scripts etc.";
+          };
+          admin = mkOption {
+            type = types.str;
+            description = "UIDs (as per the configured saml_uid_attribute) of FileSender administrators. Accounts with these UIDs can access the Admin page through the web UI. Separate multiple addresses with a comma (',').";
```

```suggestion
            description = ''
              UIDs (as per the configured saml_uid_attribute) of FileSender administrators.
              Accounts with these UIDs can access the Admin page through the web UI.
            '';
```
if github is already wrapping the line, it is a good indication that it is to long